File: SDKs/googleapis.com/identitytoolkit/v3/python/src/sdk/models/shared/createauthuriresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class CreateAuthURIResponse:
    r"""CreateAuthURIResponse
    Response of creating the IDP authentication URL.
    """
    
    all_providers: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('allProviders') }})
    auth_uri: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('authUri') }})
    captcha_required: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('captchaRequired') }})
    for_existing_provider: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('forExistingProvider') }})
    kind: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('kind') }})
    provider_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('providerId') }})
    registered: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('registered') }})
    session_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sessionId') }})
    signin_methods: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('signinMethods') }})
    

File: SDKs/googleapis.com/identitytoolkit/v3/java/lib/src/main/java/openapisdk/models/shared/CreateAuthUriResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CreateAuthUriResponse
 * Response of creating the IDP authentication URL.
**/
public class CreateAuthUriResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("allProviders")
    public String[] allProviders;
    public CreateAuthUriResponse withAllProviders(String[] allProviders) {
        this.allProviders = allProviders;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("authUri")
    public String authUri;
    public CreateAuthUriResponse withAuthUri(String authUri) {
        this.authUri = authUri;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("captchaRequired")
    public Boolean captchaRequired;
    public CreateAuthUriResponse withCaptchaRequired(Boolean captchaRequired) {
        this.captchaRequired = captchaRequired;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("forExistingProvider")
    public Boolean forExistingProvider;
    public CreateAuthUriResponse withForExistingProvider(Boolean forExistingProvider) {
        this.forExistingProvider = forExistingProvider;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("kind")
    public String kind;
    public CreateAuthUriResponse withKind(String kind) {
        this.kind = kind;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("providerId")
    public String providerId;
    public CreateAuthUriResponse withProviderId(String providerId) {
        this.providerId = providerId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("registered")
    public Boolean registered;
    public CreateAuthUriResponse withRegistered(Boolean registered) {
        this.registered = registered;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sessionId")
    public String sessionId;
    public CreateAuthUriResponse withSessionId(String sessionId) {
        this.sessionId = sessionId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("signinMethods")
    public String[] signinMethods;
    public CreateAuthUriResponse withSigninMethods(String[] signinMethods) {
        this.signinMethods = signinMethods;
        return this;
    }
}